pkg/controller/bastion/options.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardener-extension-provider-gcp/pkg/gcp"
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Options contains provider-related information required for setting up
+// a bastion instance. This struct combines precomputed values like the
+// bastion instance name with the IDs of pre-existing cloud provider
+// resources, like the VPC ID, subnet ID etc.
+type Options struct {
+	Shoot               *gardencorev1beta1.Shoot
+	BastionInstanceName string
+	FirewallName        string
+	PrivateIP           string
+	PublicIP            string
+	DiskName            string
+	UserData            string
+	Zone                string
+	Region              string
+	SubnetWork          string
+	ProjectID           string
+}
+
+// DetermineOptions determines the required information that are required to reconcile a Bastion on GCP. This
+// function does not create any IaaS resources.
+func DetermineOptions(ctx context.Context, bastion *extensionsv1alpha1.Bastion, cluster *controller.Cluster) (*Options, error) {
+	name := cluster.ObjectMeta.Name
+	bastionInstanceName := fmt.Sprintf("%s-%s-bastion", name, bastion.Name)
+	firewallName := fmt.Sprintf("%s-allow-ssh-access", bastionInstanceName)
+	diskName := fmt.Sprintf("%s-%s-disk", name, bastion.Name)
+	publicIP := bastion.Spec.Ingress[0].IPBlock.CIDR
```

Comment:
Does this mean only the first CIDR is actually allowed to reach the Bastion?

Reply:
Yes, and in bastion.yaml only one CIDR will be used per each time

Reply:
I mean only accept first CIDR. and the type of publicIP is `[]extensionsv1alpha1.BastionIngressPolicy`, and CIRD is `string`, that why []string{opt.PublicIP},

![image](https://user-images.githubusercontent.com/42234376/125384909-2cd62f80-e3cc-11eb-8bb3-43ffc8539aa1.png)

![image](https://user-images.githubusercontent.com/42234376/125385167-935b4d80-e3cc-11eb-9b9d-456d9ba2023d.png)

Reply:
@tedteng honestly I do not know your problem is. Please map ALL the ingress rules to `SourceRanges`, similar to how @xrstf mapped the ingress rules to ec2 ip permissions https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/controller/bastion/actuator_reconcile.go#L193-L240 and not only the first.

You will also have to skip the `ipv6` addresses as they are not supported by `SourceRanges` (as you can see in the api documentation). This limitation should be documented somewhere. @rfranzke any suggestions where a good place could be? In GEP-15?

Reply:
understood, I will add this limitation and api documentation url in the code as well.

Reply:
UPDATE: I talked with @rfranzke: instead of skipping the `ipv6` addresses an error should be returned. The client should make sure in this case to not set an ipv6 address for `gcp`

Reply:
committed in 207331e931675da727e95398fd32351b78dd480c